core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -304,7 +304,8 @@ private[spark] object Utils extends Logging {
           maxAttempts + " attempts!")
       }
       try {
-        dir = new File(root, namePrefix + "-" + UUID.randomUUID.toString)
+        dir = new File(root,
+          Option(namePrefix).getOrElse("spark") + "-" + UUID.randomUUID.toString)
```

Comment:
Ur, what about assert in the beginning of `createDirectory`? `createDirectory(root, null)` sounds like an invalid invocation.

Reply:
This code is used for non-shuffle files, isn't it?